src/internal/observable/from.ts

```diff
@@ -1,8 +1,18 @@
+import { subscribeToArray } from '../util/subscribeToArray';
+import { isArrayLike } from '../util/isArrayLike';
+import { isPromise } from '../util/isPromise';
 import { Observable } from '../Observable';
-import { ObservableInput, ObservedValueOf } from '../types';
-import { innerFrom } from './innerFrom';
+import { ObservableInput, ObservedValueOf, ReadableStreamLike } from '../types';
+import { isInteropObservable } from '../util/isInteropObservable';
+import { isAsyncIterable } from '../util/isAsyncIterable';
+import { createInvalidObservableTypeError } from '../util/throwUnobservableError';
+import { isIterable } from '../util/isIterable';
+import { isReadableStreamLike, readableStreamLikeToAsyncGenerator } from '../util/isReadableStreamLike';
+import { Subscriber } from '../Subscriber';
+import { isFunction } from '../util/isFunction';
+import { reportUnhandledError } from '../util/reportUnhandledError';
+import { observable as Symbol_observable } from '../symbol/observable';
 
-export function from<O extends ObservableInput<any>>(input: O): Observable<ObservedValueOf<O>>;
 /**
  * Creates an Observable from an Array, an array-like object, a Promise, an iterable object, or an Observable-like object.
```

Comment:
I think we should cover all [ObservableInput](https://rxjs.dev/api/index/type-alias/ObservableInput) here. Also, `an Array` and `an array-like` are the same I guess, so I think that these can be merged to one.

If this is not relevant for this PR, we can fix it in another one. If you decide to include such a fix here, please update `@param input` on the bottom of the JSDoc as well.